lib/puppet/util/windows/principal.rb

```diff
@@ -87,6 +87,10 @@ def self.lookup_account_name(system_name = nil, account_name)
 
     def self.lookup_account_sid(system_name = nil, sid_bytes)
```

Comment:
All of the bad cases will be trapped by the first call which converts it to a wide string. Even an empty string will still be two bytes long due to the encoding.  The same issue of zero length pointers will not occur.